shared/chat/conversation/input-area/normal/typing/index.js

```diff
@@ -0,0 +1,136 @@
+// @flow
+import * as React from 'react'
+import * as I from 'immutable'
+import * as Kb from '../../../../../common-adapters'
+import * as Styles from '../../../../../styles'
+
+const Names = ({names}: {names: I.Set<string>}) => {
+  const textType = Styles.isMobile ? 'BodyTinySemibold' : 'BodySmallSemibold'
+  let ret
+  switch (names.size) {
+    case 0:
+      ret = ''
+      break
+    case 1:
+      ret = (
+        <>
+          <Kb.Text key={0} type={textType}>
+            {names.first()}
+          </Kb.Text>
+          {' is typing'}
+        </>
+      )
+      break
+    case 2:
+      ret = (
+        <>
+          <Kb.Text key={0} type={textType}>
+            {names.first()}
+          </Kb.Text>
+          {' and '}
+          <Kb.Text key={1} type={textType}>
+            {names.skip(1).first()}
+          </Kb.Text>
+          {' are typing'}
+        </>
+      )
+      break
+    case 3:
+      ret = (
+        <>
+          <Kb.Text key={0} type={textType}>
+            {names.first()}
+          </Kb.Text>
+          {', '}
+          <Kb.Text key={1} type={textType}>
+            {names.skip(1).first()}
+          </Kb.Text>
+          {', and '}
+          <Kb.Text key={1} type={textType}>
+            {names.skip(2).first()}
+          </Kb.Text>
+          {' are typing'}
+        </>
+      )
+      break
+    default:
+      ret = 'multiple people are typing'
+  }
+  return ret
+}
+
+type Props = {
```

Comment:
should be exact

Reply:
I'm not sure how to do that and make the stories happy. The component has exactly `names` and the connected component has exactly `conversationIDKey`. Do you have any suggestions?

Reply:
I put in a commit to fix it.